### PR TITLE
feature[next]: support for Python 3.11

### PR DESCRIPTION
**Pull request by havogt**
Fixes hidden bugs in datamodels and extended_typing to support Python 3.11.

Actual bug fixes:
- Previous fix to support `typing.Any` implementation as a class (https://github.com/python/cpython/commit/5a4973e29f2f5c4ee8c086f40325786c62381540) didn't work in 3.11. 
- Partially concretization of generic datamodels replacing typevars was broken.
- Partially concretization of generic datamodels leaving some parameters as typevars was broken.

Other changes:
- Add python 3.11 as supported version.
- Remove dead code in comments.
- Fix some imports style to comply with our coding guidelines.

### CODE REVIEW

File: src/gt4py/eve/datamodels/core.py [hunk 1]
```diff
@@ -1348,9 +1358,14 @@ def _make_concrete_with_cache(
...
         class_name = f"{datamodel_cls.__name__}__{'_'.join(arg_names)}"
 
+    bound_type_params = {
+        tp_var.__name__: type_params_map[tp_var] for tp_var in datamodel_cls.__parameters__
+    }
+
     namespace = {
         "__annotations__": new_annotations,
         "__module__": module if module else datamodel_cls.__module__,
+        "__bound_type_params__": bound_type_params,  # TODO(havogt) is this useful information?
```
Comment (havogt) [inline src/gt4py/eve/datamodels/core.py:1368]: not sure if it's useful to store this, it allows to do the check above

File: src/gt4py/eve/type_validation.py [hunk 2]
```diff
@@ -358,6 +358,8 @@ def make_is_instance_of(name: str, type_: type) -> FixedTypeValidator:
         """Create an ``FixedTypeValidator`` validator for a specific type."""
 
         def _is_instance_of(value: Any, **kwargs: Any) -> None:
+            if type_ is Any:
```
Comment (havogt) [inline src/gt4py/eve/type_validation.py:361]: possibly tests are wrong

Comment (egparedes) [reply src/gt4py/eve/type_validation.py]: The actual reason for this was actually a bug in a previous fix for dealing with the implementation of `typing.Any` as a type in modern python & typing_extensions versions.

**Comment by egparedes** [comment]
@havogt could you take a look to my changes? Should we open a new PR for you being able to review it?

**Comment by havogt** [comment]
Please open a new PR, I don't always want to get github credit for your work. ;)

**Comment by egparedes** [comment]
Clsoing inf

**Pull request closed without merging**